### PR TITLE
add catalan data structures book

### PR DESCRIPTION
**Pull request by Pablito2020**
## What does this PR do?
Add resource(s)

### Description
This pull request adds a book about datastructures and algorithms created by some professors of the University of Lleida.

### Why is this valuable (or not)?
Could be valuable for people trying to understand some data structures like graphs, linkedlists, etc and they want a book about this topic that is written in catalan.

### How do we know it's really free?
It is licensed behind a Creative Commons license, which has:
Attribution, Non commercial and no derivatives.

### For book lists, is it a book? For course lists, is it a course? etc.
This is a book (has an ISBN:  978-84-9144-101-4)

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

### CODE REVIEW

**Comment by Pablito2020** [comment]
I am participating on HacktoberFest, so I would like to have the "Hacktoberfest-accepted"  label if it is possible 😄 .

**Comment by eshellman** [comment]
language code is `ca`. 
You want TWO index headings for one resource? 
It's usually not helpful to add a category for a single resource. Try to find another category, or see if there are other books in our lists that would fit into your category.

**Comment by Pablito2020** [comment]
Hi @eshellman! The commit  5d51b3deac574f1e7aad55f7219acc8fae136570 fixes the issues you commented in the pull request review 😃 .

The two books I've added one is a book-wiki about C, and the other is licensed with a creative commons with non commercial derivatives.

Cheers!

**Comment by Pablito2020** [comment]
Some maintainer will review again this pull request? 7 days have passed and the project is accepting other pull requests. Should I change something more?

**Comment by eshellman** [comment]
you have **not** fixed the language code. it should be `ca`, not `es_CA`

**Comment by eshellman** [comment]
see #9109

File: README.md [hunk 1]
```diff
@@ -87,6 +87,7 @@ This project lists books and other resources grouped by genres:
 + [Burmese / မြန်မာဘာသာ](books/free-programming-books-my.md)
 + [Chinese / 中文](books/free-programming-books-zh.md)
 + [Czech / čeština / český jazyk](books/free-programming-books-cs.md)
++ [Catalan / catalan/ català](books/free-programming-books-es_CA.md)
```
Comment (eshellman) [inline README.md:90]:
```suggestion
+ [Catalan / catalan/ català](books/free-programming-books-ca.md)
```

**Pull request merged**